src/main/java/controller/LadderGameController.java

```diff
@@ -0,0 +1,138 @@
+package controller;
+
+import domain.*;
+import view.InputView;
+import view.OutputView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static exception.ErrorMessage.USER_NAME_NOT_EXISTS_IN_USERS_EXCEPTION;
+
+public class LadderGameController {
+
+    private List<String> firstUserPositions;
+    private List<String> lastUserPositions;
+    private List<String> resultPositions;
+
+    public void run() {
+        Users users = generateUsers();
+        Results results = generateResults(users);
+        Ladder ladder = generateLadder(users);
+        printLadder(users, results, ladder);
+
+        storeLadderGameResult(users, results, ladder);
+
+        printResult(users);
+    }
+
+    private void printResult(Users users) {
+        String name = InputView.readTargetUserName();
+        while(!"all".equals(name)) {
+            printResultWhenTargetIsNotAll(users, name);
+            name = InputView.readTargetUserName();
+        }
+        printAllResult();
+    }
+
+    private Users generateUsers() {
+        Users users = null;
+        try {
+            users = new Users(InputView.readUserNames());
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            generateUsers();
+        }
+        return users;
+    }
```

Comment:
```suggestion
    private Users generateUsers() {
        try {
            return new Users(InputView.readUserNames());
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
            return generateUsers();
        }
    }
```
null 로 변수를 초기화하는 방법은 최대한 피하는게 좋을것 같아요

Reply:
아 이렇게 하면 되겠군요ㅎㅎ
감사합니다!!